src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/perform_interaction_test.cc

```diff
@@ -254,6 +257,49 @@ TEST_F(PerformInteractionRequestTest,
                             "UI is not supported by system");
 }
 
+TEST_F(PerformInteractionRequestTest, ChoiceProcessesOnHHMI_REJECT) {
+  MessageSharedPtr message = CreateMessage(smart_objects::SmartType_Map);
+  (*message)[strings::params][strings::connection_key] = kConnectionKey;
+  (*message)[strings::msg_params][strings::interaction_choice_set_id_list][0] =
+      kChoiceSetId;
+  smart_objects::SmartObject* choice_set_id =
+      &((*message)[am::strings::msg_params]
+                  [am::strings::interaction_choice_set_id]);
+  std::shared_ptr<PerformInteractionRequest> command =
+      CreateCommand<PerformInteractionRequest>(message);
+
+  smart_objects::SmartObject& response = *message;
+
+  response[strings::params][strings::message_type] = am::MessageType::kResponse;
+  response[strings::params][strings::correlation_id] = kCorrelationId;
+  response[strings::params][strings::protocol_type] = 0;
+  response[strings::params][strings::protocol_version] = 3;
+  response[strings::params][strings::connection_key] = kConnectionKey;
+  response[strings::params][strings::function_id] =
+      mobile_apis::FunctionID::PerformInteractionID;
+  response[strings::msg_params][strings::success] = false;
+  response[strings::msg_params][strings::result_code] =
+      mobile_apis::Result::REJECTED;
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillRepeatedly(Return(mock_app_));
+  {
+    InSequence seq;
```

Comment:
@IGapchuk any reason of strict sequencing?

Reply:
@AKalinich-Luxoft strict sequencing needed to exactly check behavior of PerformInteractionRequest.